codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -253,6 +253,30 @@
     }
 
     /**
+     * A key to be used for associating application-defined properties with streams within this connection.
+     *
+     * @param <K> The object type underlying the key.
+     * @param <V> The value type to be stored by this key.
+     */
+    interface PropertyKey<K, V> {
```

Comment:
If I remove the `V` parameter, then we have one of the following options for what the `getProperty` method will look like:

```java
// Option 1
MyClass value = stream.<MyClass>getProperty(key);

// Option 2
MyClass value = (MyClass)stream.getProperty(key);

// Option 3
MyClass value = stream.getProperty(key, MyClass.class);
```